.github/workflows/github-actions-ci.yml

```diff
@@ -1,68 +1,160 @@
-
-      # - name: Cocoapods - macOS Empty project
-      #   run: set -o pipefail && cd TestProjects/cocoapods/macos && arch -x86_64 pod install && xcodebuild build -workspace PinLayout-macOS.xcworkspace -scheme PinLayout-macOS -sdk macosx12.1 | xcpretty && cd ../../.. 
-
-      # - name: Cocoapods - tvOS Empty project
-      #   run: set -o pipefail && cd TestProjects/cocoapods/tvos && arch -x86_64 pod install && xcodebuild build -workspace PinLayout-tvOS.xcworkspace -scheme PinLayout-tvOS -sdk appletvsimulator16.1 -destination 'platform=tvOS Simulator,name=Apple TV 4K (2nd generation),OS=16.1' | xcpretty && cd ../../.. 
-
-      # - name: Carthage - iOS Empty project
-      #   run: set -o pipefail && cd TestProjects/carthage/ios && rm Cartfile && echo "git \"file:///$BASEDIR\"" > Cartfile && carthage update --use-ssh --platform iOS --use-xcframeworks && xcodebuild build -project PinLayout-Carthage-iOS.xcodeproj -scheme PinLayout-Carthage-iOS -sdk iphonesimulator16.2 -destination 'platform=iOS Simulator,name=iPhone SE (3rd generation),OS=16.2' | xcpretty && rm Cartfile.resolved && cd ../../.. 
+        run: |
+          set -o pipefail && xcodebuild build \
+            -project PinLayout.xcodeproj \
+            -scheme PinLayout-macOS \
+            -sdk macosx \
+            | xcpretty
+
+      - name: Build PinLayoutSample
+        run: |
+          set -o pipefail && xcodebuild build \
+            -workspace PinLayout.xcworkspace \
+            -scheme PinLayoutSample \
+            -sdk iphonesimulator \
+            -destination "platform=iOS Simulator,name=${{ env.IOS_SIMULATOR_DEVICE }},OS=${{ env.IOS_SIMULATOR_OS }}" \
+            | xcpretty
+
+      - name: iOS Unit Tests
+        run: |
+          set -o pipefail && xcodebuild build test \
+            -workspace PinLayout.xcworkspace \
+            -scheme PinLayout-iOS \
+            -sdk iphonesimulator \
+            -destination "platform=iOS Simulator,name=${{ env.IOS_SIMULATOR_DEVICE }},OS=${{ env.IOS_SIMULATOR_OS }}" \
+            | xcpretty
+
+      # - name: tvOS Unit Tests
+      #   run: |
+      #     set -o pipefail && xcodebuild build test \
+      #       -workspace PinLayout.xcworkspace \
+      #       -scheme PinLayout-tvOS \
+      #       -sdk appletvsimulator \
+      #       -destination "platform=tvOS Simulator,name=${{ env.TVOS_SIMULATOR_DEVICE }},OS=${{ env.TVOS_SIMULATOR_OS }}" \
+      #       | xcpretty
+      
+      # - name: macOS Unit Tests
+      #   run: |
+      #     set -o pipefail && xcodebuild test \
+      #       -workspace PinLayout.xcworkspace \
+      #       -scheme PinLayout-macOS \
+      #       -sdk macosx \
+      #       -destination "platform=macOS,name=Any Mac" \
+      #       | xcpretty
```

Comment:
Test compilation fails with the current project structure. For now, I have fixed the existing test code to make it work, but further improvements should be addressed in a separate PR.